modules/common/services/storage.nix

```diff
@@ -0,0 +1,114 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{lib, ...}: let
+  inherit (lib) mkEnableOption mkOption types literalExpression;
+
+  storageSubmodule = types.submodule {
+    options = {
+      tag = mkOption {
+        type = types.nullOr types.str;
+        description = ''
+          Storage tag used as label for the microvm share.
+        '';
+      };
+      host-path = mkOption {
+        type = types.str;
+        description = ''
+          Storage directory in the host. If it does not exist, it will be generated
+          with the default owner 'microvm' and group 'kvm'.
+        '';
+      };
+      tmp-path = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          (Optional) Temporary directory in the VM to mount the host directory. If this is set,
+          the folder from host-path will be recursively copied from 'tmp-path' to 'vm-path' and owner, group,
+          and permissions applied.
+          If this parameter is null, the host-path is mounted to vm-path directly.
+        '';
+      };
+      vm-path = mkOption {
+        type = types.str;
+        description = ''
+          Storage directory in the guest where the share is mounted to.
+        '';
+      };
+      target-vm = mkOption {
+        type = types.str;
+        description = ''
+          Name of the VM the share is mounted into. The name must match the microvm
+          name in 'config.microvm.vms': e.g., "gui-vm" (note the '-').
```

Comment:
You can use `assertions = [{}];` nixos option and `config.microvm.vms ? ${target-vm}` condition here to test if the microvm actually exists to check for misconfiguration.

Reply:
Yes will add assertions later if this is going to be used